ui/webui/resources/wasm/opaque_ke/src/lib.rs

```diff
@@ -34,41 +34,35 @@ pub struct Registration {
 #[wasm_bindgen]
 impl Registration {
     #[wasm_bindgen(constructor)]
-    pub fn new() -> Registration {
-        Registration {
+    pub fn new() -> Self {
+        Self {
             state: None,
             rng: OsRng
         }
     }
 
     // result => POST /v2/accounts/password/init
     pub fn start(&mut self, password: &str) -> Result<String, JsValue> {
-        match ClientRegistration::<DefaultCipherSuite>::start(
+        let result = ClientRegistration::<DefaultCipherSuite>::start(
             &mut self.rng, &password.as_bytes()
-        ) {
-            Ok(result) => {
-                self.state = Some(result.state);
-                return Ok(hex::encode(result.message.serialize()));
-            },
-            Err(_error) => return Err(
-                "ClientRegistration::<DefaultCipherSuite>::start() failed!".into()
-            )
-        };
+        ).map_err(|_| "ClientRegistration::<DefaultCipherSuite>::start() failed!")?;
+
+        self.state = Some(result.state);
+        Ok(hex::encode(result.message.serialize()))
     }
 
     // result => POST /v2/accounts/password/finalize
     pub fn finish(&mut self, hex_response: &str, password: &str, email: &str) -> Result<String, JsValue> {
-        let serialized_response = match hex::decode(&hex_response) {
-            Ok(serialized_response) => serialized_response,
-            Err(_error) => return Err("hex::decode() failed!".into())
-        };
-
-        let response = match RegistrationResponse::deserialize(&serialized_response) {
-            Ok(response) => response,
-            Err(_error) => return Err("RegistrationResponse::deserialize() failed!".into())
-        };
+        if self.state.is_none() {
+            return Err("Calling Registration::finish() \
+                        without first calling Registration::start()!".into());
+        }
 
-        match self.state.take().unwrap().finish(
+        let serialized_response = hex::decode(&hex_response)
+            .map_err(|_| "hex::decode() failed!")?;
+        let response = RegistrationResponse::deserialize(&serialized_response)
+            .map_err(|_| "RegistrationResponse::deserialize() failed!")?;
+        let result = self.state.take().unwrap().finish(
```

Comment:
optional nit: on second thought, you could use `take().ok_or()?` instead of using a check at the top, but no biggie

Reply:
https://github.com/brave/brave-core/pull/28243/commits/f0c249da34e32979c5fae9106d0de56a20f1f82a